core/embed/boardloader/main.c

```diff
@@ -160,6 +162,18 @@ static uint32_t check_sdcard(void) {
       return 0;
     }
 
+    if (sectrue != (check_single_hash(
```

Comment:
we should static_assert something like `IMAGE_CHUNK_SIZE >= BOOTLOADER_IMAGE_MAXSIZE` so that we can be sure that the bootloader is covered by exactly one hash, right?

(also technically we should check that all the other hashes are zero. could we reuse `check_image_contents`?

Reply:
added check of other hashes and static assert: https://github.com/trezor/trezor-firmware/commit/b4a3ad09e9a2c1306683c2a07606eea491875c6d